addons/l10n_fi/models/account_move.py

```diff
@@ -0,0 +1,93 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import re
+from odoo import models, _
+from odoo.exceptions import UserError
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def number2numeric(number):
+
+    invoice_number = re.sub(r'\D', '', number)
+
+    if invoice_number == '' or invoice_number is False:
+        raise UserError(_('Invoice number must contain numeric characters'))
+
+    # Make sure the base number is 3...19 characters long
+    if len(invoice_number) < 3:
+        log.warning('Invoice number {} is less than 3 characters long, '
+                    'padding to 3 characters with prefix 11.'
+                    .format(invoice_number))
+        invoice_number = ('11' + invoice_number)[-3:]
+    elif len(invoice_number) > 19:
+        log.warning('Invoice number {} is over 19 characters long, '
+                    'truncating to 19 characters'.format(invoice_number))
+        invoice_number = invoice_number[:19]
+
+    return invoice_number
```

Comment:
I don't see why these methods should be outside of the class.  The number2numeric warnings are an exaggeration to me (they are not in the other localizations and I don't see why FI should be an exception)

Reply:
- Warnings removed
- moved methods to class and set them static (except from one to classmethod)